src/audio_unit/looper.rs

```diff
@@ -6,13 +6,77 @@ use State::*;
 #[derive(Debug)]
 pub enum Message {
     Toggle,
+    QueueOverdub,
 }
 
 #[derive(Debug)]
 enum State {
     Off,
     Recording { position: usize },
     Playing { position: usize, total: usize },
+    PlayingAwaitingOverdub { position: usize, total: usize },
+    Overdubbing { position: usize, total: usize },
+}
+
+impl State {
+    pub fn is_awaiting_overdub(&self) -> bool {
+        matches!(self, State::PlayingAwaitingOverdub { .. })
```

Comment:
Per [this answer](https://stackoverflow.com/a/51123901/732366) (for checking whether an enum value is a certain variant)

Reply:
Ooh nice, I haven't seen that before.